library/common/extensions/filters/http/platform_bridge/filter.cc

```diff
@@ -74,26 +73,41 @@ PlatformBridgeFilter::PlatformBridgeFilter(PlatformBridgeFilterConfigSharedPtr c
   if (platform_filter_.init_filter == nullptr) {
     ENVOY_LOG(debug, "PlatformBridgeFilter({}): missing initializer", filter_name_);
     platform_filter_ = {};
+
+    // Does this make sense, or should we rather protect the rest of the code against nullptr for
+    // both request_filter_base_ and response_filter_base_.
+    request_filter_base_ = std::make_unique<RequestFilterBase>(*this);
+    response_filter_base_ = std::make_unique<ResponseFilterBase>(*this);
+
     return;
   }
 
   // Set the instance_context to the result of the initialization call. Cleanup will ultimately
   // occur within the onDestroy() invocation below.
   ENVOY_LOG(trace, "PlatformBridgeFilter({})->init_filter", filter_name_);
-  platform_filter_.instance_context = platform_filter_.init_filter(platform_filter_.static_context);
+  platform_filter_.instance_context = platform_filter_.init_filter(&platform_filter_);
   ASSERT(platform_filter_.instance_context,
          fmt::format("PlatformBridgeFilter({}): init_filter unsuccessful", filter_name_));
+
+  // Set directional filters now that the platform filter has been init'ed.
+  request_filter_base_ = std::make_unique<RequestFilterBase>(*this);
+  response_filter_base_ = std::make_unique<ResponseFilterBase>(*this);
 }
 
 void PlatformBridgeFilter::setDecoderFilterCallbacks(
     Http::StreamDecoderFilterCallbacks& callbacks) {
   ENVOY_LOG(trace, "PlatformBridgeFilter({})::setDecoderCallbacks", filter_name_);
   decoder_callbacks_ = &callbacks;
 
+  // TODO(goaway): currently both platform APIs unconditionally set this field, meaning that the
+  // heap allocation below occurs when it could be avoided.
   if (platform_filter_.set_request_callbacks) {
     platform_request_callbacks_.resume_iteration = envoy_filter_callback_resume_decoding;
     platform_request_callbacks_.release_callbacks = envoy_filter_release_callbacks;
     // We use a weak_ptr wrapper for the filter to ensure presence before dispatching callbacks.
+    // Note that the weak_ptr here is heap allocated because it needs to be managed and held by
+    // platform specific runtimes. Therefore, it is NECESSARY that the platform deallocates the
+    // memory. Otherwise the weak ptr will leak.
```

Comment:
Wait, aren't we going to address this in this PR?

Reply:
Yeah, it is addressed in the leaking case. But I wanted to leave the comment, because it is the case that if the pointer gets allocated it is up to the platform code to release.

Reply:
Okay, sure, we can leave the comment for clarity. Maybe simplify to something like "The weak_ptr is heap-allocated, because it must be managed (and eventually released) by platform code."